usecases/traverser/explorer_hybrid.go

```diff
@@ -0,0 +1,320 @@
+					if err != nil {
+						return nil, err
+					}
+				} else {
+					params.SearchVector = params.HybridSearch.Vector
+				}
+
+				// Build a new vearvec search
+				nearVecParams := &searchparams.NearVector{
+					Vector:        params.SearchVector,
+					TargetVectors: params.HybridSearch.TargetVectors,
+				}
+
+				res, name, err := denseSearch(ctx, nearVecParams, e, params, "hybridVector")
+				if err != nil {
+					e.logger.WithField("action", "hybrid").WithError(err).Error("denseSearch failed")
+					return nil, err
+				} else {
+					weights = append(weights, params.HybridSearch.Alpha)
+					results = append(results, res)
+					names = append(names, name)
+				}
+			}
+		}
+	}
+
+	// If the user has given any weight to the keyword search, do a keyword search
+	if 1-params.HybridSearch.Alpha > 0 {
+		sparseResults, name, err := sparseSearch(ctx, e, params)
+		if err != nil {
+			e.logger.WithField("action", "hybrid").WithError(err).Error("sparseSearch failed")
+			return nil, err
+		} else {
+			weights = append(weights, 1-params.HybridSearch.Alpha)
+			results = append(results, sparseResults)
+			names = append(names, name)
+		}
+	}
+
+	// The postProcess function is used to limit the number of results and to resolve references
+	// in the results.  It is called after all the subsearches have been completed, and before autocut
+	postProcess := func(results []*search.Result) ([]search.Result, error) {
+		totalLimit, err := e.CalculateTotalLimit(origParams.Pagination)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(results) > totalLimit {
+			results = results[:totalLimit]
+		}
+
+		res1 := make([]search.Result, 0, len(results))
+		for _, res := range results {
+			res1 = append(res1, *res)
+		}
+
+		res, err := e.searcher.ResolveReferences(ctx, res1, origParams.Properties, nil, origParams.AdditionalProperties, origParams.Tenant)
+		if err != nil {
+			return nil, err
+		}
+		return res, nil
+	}
+
+	res, err := hybrid.HybridSubsearch(ctx, &hybrid.Params{
+		HybridSearch: origParams.HybridSearch,
+		Keyword:      origParams.KeywordRanking,
+		Class:        origParams.ClassName,
+		Autocut:      origParams.Pagination.Autocut,
+	}, results, weights, names, e.logger, postProcess)
+	if err != nil {
+		return nil, err
+	}
+
+	var pointerResultList hybrid.Results
+
+	if origParams.Pagination.Limit <= 0 {
+		origParams.Pagination.Limit = hybrid.DefaultLimit
+	}
+
+	if origParams.Pagination.Offset < 0 {
+		origParams.Pagination.Offset = 0
+	}
+
+	if len(res) >= origParams.Pagination.Limit+origParams.Pagination.Offset {
+		pointerResultList = res[origParams.Pagination.Offset : origParams.Pagination.Limit+origParams.Pagination.Offset]
+	}
+	if len(res) < origParams.Pagination.Limit+origParams.Pagination.Offset && len(res) > origParams.Pagination.Offset {
+		pointerResultList = res[origParams.Pagination.Offset:]
+	}
+	if len(res) <= origParams.Pagination.Offset {
+		pointerResultList = hybrid.Results{}
+	}
```

Comment:
why are we doing this? Comment?

Reply:
I have no idea, left over from a refactor.  We can trim this one

Reply:
ah, the rest of the program still uses []search.Result and not []*search.Result.  So we have to change the type.

It's not great, but the rest of weaviate needs to change

Reply:
ah I see - could you add a comment to make this more clear?

Reply:
will do